### PR TITLE
add login integrate API

### PR DESCRIPTION
**Pull request by SoulOfDarknes**
Add login integration API

Login
1. Add token user info to persist 
2. add if user in DB return token and info about user
4. fix little bit styles

**PR checklist:**

**1.1 Common**
- [x] 1.1.2 no any, should be also added to eslint rules
- [x] 1.1.1 types for input and output params
- [x] 1.1.3 try/catch for any async function
- [x] 1.1.4 no magic numbers
- [x] 1.1.5 compare only with constants not with strings (instead of user === ‘admin’, better user === roles.admin)
- [x] 1.1.6 no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] 1.1.7 avoid similar types with duplicating by generics
- [x] 1.1.8 no console.log in pr
- [x] 1.1.9 .env file should be in .gitignore
- [x] 1.1.10 delete commented code if it's not part of documentation
- [x] 1.1.11 no links in the code, env links should be in env file (for example: server url),
- [x] constant links (for example default avatar URL) should be in constant file
- [x] 1.1.12 constants and function names should be lowercase.
- [x] 1.1.13 no dates format in the code (like "dd/MM/yyyy”), move it in global constant variable
- [x] 1.1.14 import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] 1.1.15 strings should be in the constant variable. Example: instead of ‘15 3 * * *’, const EACH_DAY_15h_15min = ‘15 3 * * *’

 **1.3 Frontend|Mobile**
- [x] 1.3.1 split component and logic (move logic to custom hook file)
- [x] 1.3.2 colors, font size, and font name should be in the theme or in the constants
- [x] 1.3.3 no text in the components, use i18n approach, even if you have only one language for now
- [x] 1.3.4 inline styles prohibited
- [x] 1.3.5 import should be absolute. instead of ../../../components/myComponent should be components/myComponent

image (10)

### CODE REVIEW

File: apps/medon-fe/src/pages/ResendConfirmation/index.tsx [hunk 1]
```diff
@@ -18,10 +18,13 @@ import {
 } from 'pages/ResendConfirmation/styles';
 import { SubmitSendEmail } from 'pages/ResendConfirmation/types';
 import { emailSchema } from 'validation/accountConfirmationSchema';
+import { useResendEmailMutation } from 'redux/api/authApi';
+import { toast } from 'react-toastify';
```
Comment (Mati-Pereira) [inline apps/medon-fe/src/pages/ResendConfirmation/index.tsx]: Order of import 

node_modules
absolute path

**Review by Mati-Pereira — CHANGES_REQUESTED** [review]
What I saw, it's just the order of imports

File: apps/medon-fe/src/components/LoginComponent/index.tsx [hunk 2]
```diff
@@ -9,20 +8,25 @@ import { useEffect } from 'react';
...
+  const authToken = params.get('gtoken');
   const verifyEmail = useVerifyEmailQuery(
     { token },
     {
       skip: !token,
     }
   );
+  const dispatch = useAppDispatch();
 
   const handleSubmit = (data: LoginRequest) => data;
```
Comment (skazzp) [inline apps/medon-fe/src/components/LoginComponent/index.tsx]: this function does nothing

File: apps/medon-fe/src/components/Navigation/index.tsx [hunk 3]
```diff
@@ -21,6 +29,13 @@ import { navigation } from 'utils/constants/navigation';
 export default function Navigation() {
   const { t } = useTranslation();
 
+  const user = useAppSelector(getUserSelector);
+  const dispatch = useDispatch();
+  const logout = () => {
+    dispatch(setUser(initialState.user));
+    dispatch(setToken(''));
```
Comment (skazzp) [inline apps/medon-fe/src/components/Navigation/index.tsx]: change for logout reducer, cause this 2 doesn't clean all userState

File: apps/medon-fe/src/redux/features/restoreAuth/restoreAuth.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { Dispatch } from "@reduxjs/toolkit";
+import { logout, setToken } from "redux/features/userSlice/userSlice";
+
+export const restoreAuth = () => async (dispatch: Dispatch) => {
```
Comment (skazzp) [inline apps/medon-fe/src/redux/features/restoreAuth/restoreAuth.ts]: file not needed?

**Review by skazzp — CHANGES_REQUESTED** [review]

File: apps/medon-fe/src/pages/ResendConfirmation/index.tsx (hunk 1 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/pages/ResendConfirmation/index.tsx]: changed

File: apps/medon-fe/src/components/LoginComponent/index.tsx (hunk 2 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/LoginComponent/index.tsx:28]: deleted

Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/LoginComponent/index.tsx]: deleted

File: apps/medon-fe/src/components/Navigation/index.tsx (hunk 3 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/Navigation/index.tsx]: done

File: apps/medon-fe/src/redux/features/restoreAuth/restoreAuth.ts (hunk 4 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/redux/features/restoreAuth/restoreAuth.ts]: i think yeah now its no needed

**Review by Mati-Pereira — APPROVED** [review]

**Review by skazzp — APPROVED** [review]

File: apps/medon-fe/src/main.tsx [hunk 5]
```diff
@@ -20,13 +21,15 @@ const root = ReactDOM.createRoot(
...
-        <BrowserRouter>
-          <ToastContainer autoClose={2000} />
-          <GlobalStyle />
-          <App />
-        </BrowserRouter>
-      </ThemeProvider>
+      <PersistGate loading={null} persistor={persistedStore}>
+        <ThemeProvider theme={theme}>
+          <BrowserRouter>
+            <ToastContainer autoClose={2000} />
```
Comment (lyudad) [inline apps/medon-fe/src/main.tsx]: magic number

Comment (SoulOfDarknes) [reply apps/medon-fe/src/main.tsx]: done

**Pull request merged**